lib/itamae/resource/http_request.rb

```diff
@@ -0,0 +1,29 @@
+require 'itamae'
+require 'open-uri'
+
+module Itamae
+  module Resource
+    class HttpRequest < File
+      UrlNotFoundError = Class.new(StandardError)
+
+      define_attribute :headers, type: Hash, default: {}
+      define_attribute :url, type: String
+
+      def pre_action
+        unless attributes.url
```

Comment:
`define_attribute` accepts `required` option.

```ruby
define_attribute :url, type: String, required: true
```

Instead of L13-L15, please use this.